lib/src/db/db.dart

```diff
@@ -302,12 +302,43 @@ class DatastoreDB {
 
   /// Looks up [keys] in the datastore and returns a list of [Model] objects.
   ///
+  /// Any key that is not found in the datastore will have a corresponding
+  /// value of null in the list of model objects that is returned.
+  ///
   /// For transactions, please use [beginTransaction] and call the [lookup]
   /// method on it's returned [Transaction] object.
+  ///
+  /// See also:
+  ///
+  ///  * [lookupValue], which looks a single value up by its key, requiring a
+  ///    successful lookup.
   Future<List<T>> lookup<T extends Model>(List<Key> keys) {
     return _lookupHelper<T>(this, keys);
   }
 
+  /// Looks up a single [key] in the datastore, and returns the associated
+  /// [Model] object.
+  ///
+  /// If [orElse] is specified, then it will be consulted to provide a default
+  /// value for the model object in the event that [key] was not found in the
+  /// datastore.
+  ///
+  /// If the [key] is not found in the datastore and [orElse] was not
+  /// specified, then an [ArgumentError] will be thrown.
+  Future<T> lookupValue<T extends Model>(Key key, {T orElse()}) async {
+    final List<T> values = await lookup(<Key>[key]);
+    assert(values.length == 1);
+    T value = values.single;
+    if (value == null) {
+      if (orElse != null) {
+        value = orElse();
+      } else {
+        throw ArgumentError('No value associated with key: $key');
```

Comment:
It seems more natural that this would be an Exception, not an Error.

Probably needs a custom exception for "value not found" or something like that :)

Reply:
Good call - done.

I also added a corresponding method to `Transaction` as well.